.github/workflows/verify_source_generated_changes_are_persisted.yml

```diff
@@ -0,0 +1,34 @@
+name: Verify source generator changes have been persisted
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  verify_source_generators:
+    runs-on: windows-latest
+
+    steps:
+      - name: Support longpaths
+        run: git config --system core.longpaths true
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '6.0.x'
+
+      - name: "Regenerating package versions"
+        run: .\tracer\build.ps1 BuildTracerHome
+
+      - name: "Verify no changes in generated files"
+        run: |
+          git diff --quiet -- .\tracer\src\Datadog.Trace\Generated
```

Comment:
Do we _need_ to add the generated files to git? Aren't they generated on build?

Reply:
> Aren't they generated on build?

There's two parts to that - currently, yes, they're always generated in-memory on build, and thanks to the settings we added, they're also always written to this folder on disk

>Do we _need_ to add the generated files to git? 

No, we don't _need_ to. We could exclude files in that folder from git if we want to. But IMO there's a very strong case that they _should_ be in git, and that it's beneficial.

Otherwise, we can't review _all_ the changes a PR makes, because some of them are "internal" to the compiler. For the current TagsList generator, that's probably not a big deal, but for others, it really is. 

For example, consider the IntegrationDefinitions generator I was working on. That's such a critical part of the Tracer, that I think we _definitely_ want to see when there are changes made to that file in a PR?